java/src/Ice/src/main/java/com/zeroc/IceInternal/IPEndpointI.java

```diff
@@ -176,14 +173,16 @@ public java.util.List<Connector> connectors(
   }
 
   @Override
-  public synchronized int hashCode() {
-    if (!_hashInitialized) {
-      _hashValue = 5381;
-      _hashValue = HashUtil.hashAdd(_hashValue, type());
-      _hashValue = hashInit(_hashValue);
-      _hashInitialized = true;
+  public int hashCode() {
+    int h = 5381;
+    h = HashUtil.hashAdd(h, type());
+    h = HashUtil.hashAdd(h, _host);
+    h = HashUtil.hashAdd(h, _port);
+    h = HashUtil.hashAdd(h, _connectionId);
+    if (_sourceAddr != null) {
```

Comment:
I think it would be clearer to provide and use a Network.hashAddress to do this hashing.

Keeping in mind this hash needs to be consistent with equals/compareTo, which uses Network.compareAddress.

Reply:
I was actually planning on deleting `compareAddress` in the future...

If you look at the implementation, it's just an over-complicated way to compare `getPort` and `getAddress`.
Over-complicated, because we aggressively tried to optimize it, and because that function can actually return 
`-1`, `0`, or `1` to give `<`, `==`, or `>`... but we only ever check if this value is equal to `0`.
So most of the function is useless anyways.

I'm actually like 85% sure this entire function is pointless, and we should just be using the `equals` on `INetSocketAddress`... From the docs, it sure sounds like it's doing exactly what we're doing here:
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/net/InetSocketAddress.html#equals(java.lang.Object)